### PR TITLE
support unackMessages stats for cumulative ack

### PR DESCRIPTION
**Pull request by ltamber**
### Motivation

Currently, the unackMessages was always 0 in cumulativeAck mode if show topic stats, this pull request calculated unacked messages both in cumulative ack and individual ack mode.

### CODE REVIEW

File: pulsar-broker/src/main/java/org/apache/pulsar/broker/service/Consumer.java [hunk 1]
```diff
@@ -169,12 +170,7 @@ public Consumer(Subscription subscription, SubType subType, String topicName, lo
...
         stats.metadata = this.metadata;
 
-        if (Subscription.isIndividualAckMode(subType)) {
-            this.pendingAcks = new ConcurrentLongLongPairHashMap(256, 1);
-        } else {
-            // We don't need to keep track of pending acks if the subscription is not shared
-            this.pendingAcks = null;
-        }
+        this.pendingAcks = new ConcurrentLongLongPairHashMap(256, 1);
     }
```
Comment (sijie) [inline pulsar-broker/src/main/java/org/apache/pulsar/broker/service/Consumer.java:174]: Can we avoid creating a hashmap if we don't need it?

**Review by sijie — CHANGES_REQUESTED** [review]

File: pulsar-broker/src/main/java/org/apache/pulsar/broker/service/Consumer.java (hunk 1 above)
Comment (ltamber) [reply pulsar-broker/src/main/java/org/apache/pulsar/broker/service/Consumer.java:174]: we need to track of pending acks in all subscription mode, I thinks this is also need to create a hashmap

Comment (sijie) [reply pulsar-broker/src/main/java/org/apache/pulsar/broker/service/Consumer.java:174]: pending acks are only used when the transaction is enabled.

**Comment by zymap** [comment]
Move this to the next release.

**Comment by Anonymitaet** [comment]
Hi @ltamber thanks for your code work. For this PR, do we need to update docs?

**Comment by congbobo184** [comment]
@ltamber hi, I am working on release pulsar 2.7.3, could you please solve the conflict and fix the comment? thanks.

**Comment by michaeljmarshall** [comment]
@ltamber - do you plan on completing the work for this PR? There are conflicts right now. As such, I am going to remove the 2.7.4 label. I added the 2.10 milestone. If/when we merge this, we can select the branches to cherry-pick to.

**Comment by ltamber** [comment]
> @ltamber - do you plan on completing the work for this PR? There are conflicts right now. As such, I am going to remove the 2.7.4 label. I added the 2.10 milestone. If/when we merge this, we can select the branches to cherry-pick to.

sorry for that, I'll close this PR first.

**Comment by michaeljmarshall** [comment]
@ltamber - no worries. Do you know if this issue still exists? If it does, maybe we can create a GitHub issue to track and hopefully resolve the bug. Thanks.

**Pull request closed without merging**